src/app/shared/services/blockchain/campaign/campaign.service.ts

```diff
@@ -97,6 +97,15 @@ export class CampaignService {
   }
 
   create(data: CreateCampaignData, flavor: CampaignFlavor): Observable<string | undefined> {
+    const realSoftCap = BigNumber.from(data.softCap)
```

Comment:
I suppose this is here because of the rounding errors, right?

Reply:
yes. there were issues when you want to make a campaign with data that makes every number irrational.. we need to set a soft cap that will be calculated in reverse. long story. along with filip, we made the same things on both sides to mitigate this issue.